grails-app/controllers/org/pih/warehouse/api/ReplenishmentApiController.groovy

```diff
@@ -44,10 +48,15 @@ class ReplenishmentApiController {
     }
 
     def create = {
+        Location currentLocation = Location.get(session.warehouse.id)
         JSONObject jsonObject = request.JSON
-
+        jsonObject.replenishmentItems.each { item ->
+            Product product = Product.get(item['product.id'])
+            if (item.quantity > inventoryService.getQuantityAvailableToPromise(product, currentLocation)) {
```

Comment:
I think you should compare it with `QATP - quantity in bin` plus move it to the service, to the `createOrUpdateOrderFromReplenishment`, and use `validateReplenishmentItem` method from the same service.